rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -5576,6 +5583,176 @@ public void addSignatureCreateEventLog() throws Exception {
         verify(eventLogger, times(1)).logAddSignature(federatorPubKey, btcTx, rskTxHash.getBytes());
     }
 
+    @Test
+    public void addSignature_invalidSignatureBeforeRSKIP326_eventStillEmitted() throws Exception {
```

Comment:
I like the idea you proposed about creating an auxiliary test method to avoid repeating the same logic in every test.

It could be something like this 
```
private void test_addSignature_EventEmmited(boolean isRskip326Active, boolean isSignatureValid, boolean federatorAlreadySigned) {
```
That would allow us to test different cases with the same code I think